cdap-ui/app/cdap/components/FieldLevelLineage/v2/Context/FllContext.tsx

```diff
@@ -45,40 +52,52 @@ export interface IContextState {
   causeSets: ITableFields;
   impactSets: ITableFields;
   showingOneField: boolean;
+  start: number;
+  end: number;
+  selection: string;
   handleFieldClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
   handleViewCauseImpact?: () => void;
   handleReset?: () => void;
-  activeField?: string;
+  activeField?: IField;
   activeCauseSets?: ITableFields;
   activeImpactSets?: ITableFields;
   activeLinks?: ILink[];
   numTables?: number;
   firstCause?: number;
   firstImpact?: number;
   firstField?: number;
+  setTimeRange?: (range: string) => void;
+  setCustomTimeRange?: ({ start, end }) => void;
 }
 
 export class Provider extends React.Component<{ children }, IContextState> {
   private handleFieldClick = (e) => {
-    const activeField = e.target.id;
-    if (!activeField) {
+    const activeFieldId = e.target.id;
+    if (!activeFieldId) {
       return;
     }
-    d3.select(`#${this.state.activeField}`).classed('selected', false);
+    if (this.state.activeField) {
+      d3.select(`#${this.state.activeField.id}`).classed('selected', false);
+    }
 
+    const newField = {
+      id: activeFieldId,
+      name: e.target.dataset.fieldname,
+    };
     this.setState(
       {
-        activeField,
-        activeLinks: this.getActiveLinks(),
+        activeField: newField,
+        activeLinks: this.getActiveLinks(activeFieldId),
       },
       () => {
-        d3.select(`#${activeField}`).classed('selected', true);
+        d3.select(`#${activeFieldId}`).classed('selected', true);
+        replaceHistory(this);
```

Comment:
Here too. Do not pass the `this` object around. This defeats the purpose of having a Class in javascript and having everything related under the class umbrella.

Reply:
Refactored to avoid passing in Context component -